device/tt_soc_descriptor.h

```diff
@@ -123,24 +123,35 @@ class tt_SocDescriptor {
     std::vector<std::size_t> trisc_sizes;  // Most of software stack assumes same trisc size for whole chip..
     std::string device_descriptor_file_path = std::string("");
     bool has(tt_xy_pair input) { return cores.find(input) != cores.end(); }
-    int overlay_version;
-    int unpacker_version;
     int dst_size_alignment;
-    int packer_version;
     int worker_l1_size;
     int eth_l1_size;
     bool noc_translation_id_enabled;
     uint64_t dram_bank_size;
 
+    // Vectors used to translate logical coordinates to physical and translated coordinates.
+    std::map<std::size_t, std::size_t> physical_y_to_logical_y;
+    std::map<std::size_t, std::size_t> physical_x_to_logical_x;
```

Comment:
do you think the maps could be changed to vectors for faster look ups?

Reply:
In theory I could, but physical coordinates are not continuous in general, like logical. That would make some vector elements unused, which is not a big problem but I think with map it is more clear that physical coordinates are not continuous. I get that you are considered for perf, but I think this is not going to have a big impact. Map is going to be small and it is using integers as keys so it shouldn't be a problem.
However, I want to confirm this so I could write a small test on a separate branch with both vector and map to confirm that perf penalty is not big / doesn't exist

Reply:
I think that should be okay, you are right in that there are gaps and handling that would be awkward.